File: legacy/packages/contracts/python/src/gta_ai_contracts/quality.py
from __future__ import annotations

from typing import Literal
from uuid import UUID

from pydantic import Field

from .common import JsonValue, StudioModel, UnitScore


class QualityCheckResult(StudioModel):
    id: UUID
    render_job_id: UUID
    check_id: str
    check_version: str
    dimension: Literal["technical", "editorial", "factual", "audio", "subtitle", "visual", "platform"]
    status: Literal["pass", "warn", "fail", "skipped"]
    severity: Literal["info", "warning", "blocker"]
    message: str
    measured_value: JsonValue
    threshold: JsonValue
    evidence_artifact_ids: list[UUID] = Field(default_factory=list)
    correction_action: str | None = None


class QualityScore(StudioModel):
    editorial_adherence: UnitScore
    factual_reliability: UnitScore
    visual_quality: UnitScore
    pacing: UnitScore
    audio_quality: UnitScore
    subtitle_quality: UnitScore
    platform_compliance: UnitScore
    overall: UnitScore


class QualityGateDecision(StudioModel):
    passed: bool
    blocker_check_ids: list[str] = Field(default_factory=list)
    warning_check_ids: list[str] = Field(default_factory=list)
    score: QualityScore

File: legacy/packages/contracts/src/quality.ts
import type { JsonValue, Uuid } from "./common.js";

export type QualityDimension = "technical" | "editorial" | "factual" | "audio" | "subtitle" | "visual" | "platform";
export type QualityCheckStatus = "pass" | "warn" | "fail" | "skipped";
export type QualitySeverity = "info" | "warning" | "blocker";

export interface QualityCheckResult {
  readonly id: Uuid;
  readonly render_job_id: Uuid;
  readonly check_id: string;
  readonly check_version: string;
  readonly dimension: QualityDimension;
  readonly status: QualityCheckStatus;
  readonly severity: QualitySeverity;
  readonly message: string;
  readonly measured_value: JsonValue;
  readonly threshold: JsonValue;
  readonly evidence_artifact_ids: readonly Uuid[];
  readonly correction_action: string | null;
}

export interface QualityScore {
  readonly editorial_adherence: number;
  readonly factual_reliability: number;
  readonly visual_quality: number;
  readonly pacing: number;
  readonly audio_quality: number;
  readonly subtitle_quality: number;
  readonly platform_compliance: number;
  readonly overall: number;
}

export interface QualityGateDecision {
  readonly passed: boolean;
  readonly blocker_check_ids: readonly string[];
  readonly warning_check_ids: readonly string[];
  readonly score: QualityScore;
}

